vaxrank/vaccine_peptide.py

```diff
@@ -54,14 +60,35 @@ def lexicographic_sort_key(self):
         score positive (since we want fewer wildtype epitopes) but the others
         negative (since we want more of them).
         """
-        return (
+        # since we're sorting in decreasing order, numbers which we want
+        # to be larger must have their signs flipped
+        essential_score_tuple = (
+            # Sum of normalized MHC binding affinities of subsequences
             -self.mutant_epitope_score,
-            -self.mutant_protein_fragment.n_alt_reads,
+            # Number of reads supporting the variant
+            -self.mutant_protein_fragment.n_alt_reads
+        )
+        manufacturability_score_tuple = tuple(self.manufacturability_scores)
+        extra_score_tuple = (
+            # Number of reads supporting the particular protein sequence
+            # sequence we're using for this vaccine peptide. Currently
+            # all vaccine peptides are drawn from the same larger sequence
+            # so this score shouldn't change.
             -self.mutant_protein_fragment.n_alt_reads_supporting_protein_sequence,
+            # Minimize the sum of non-mutant MHC binding scores
             self.wildtype_epitope_score,
+            # All else being equal, we prefer to maximize the number of
+            # mutant amino acids
             -self.mutant_protein_fragment.n_mutant_amino_acids,
+            # If nothing else can serve as a tie break then try to center
+            # the mutation in the vaccine peptide.
             -self.mutant_protein_fragment.mutation_distance_from_edge
         )
+        return (
+            essential_score_tuple +
+            manufacturability_score_tuple +
```

Comment:
Hey, 

I'll think about this and get back to you. I would lean towards warning the end-user about manufacturability concerns (e.g. "WARNING - Very Hydrophobic C-terminus"), and suggesting alternatives rather than having "manufacturability" factor into ranking. Peptide synthesis is an evolving science, and you probably don't not want to bake-in current synthesis limitations. For example, the reactive residues JPT suggests we avoid (e.g. Cysteine) are useful handles for creating peptide-conjugates, which many groups are exploring.